### DISCUSSION TITLE
Get progress for all nodes for an API, not just Progress on Image creation

### DISCUSSION THREAD

**Discussion by rawmen0x** [Comfy-Org/ComfyUI#1767] [Q&A]
Hi,

I'm trying to get the Overall progress on a Prompt. I noticed that hijack progress on main script only gets the progress on Ksamplers nodes.

Is there any util function that gets the overall progress for all nodes in a prompt?

On events I see Progress only for the image creation event I mentioned, I want to check all nodes including the ones marked on events as Executing.

Thank you.

**Comment by Qbsoon** [comment]
Having same issue. Only shows on one ksampler and so the progress is shown wrong when there's 2 samplers for example, its shows progress for the 1st one, then for 2nd one instead of showing full progress for the whole workflow sent to the server.
